File: LeetCode/cpp/682_Baseball_Game.cpp
class Solution {
public:
	int calPoints(vector<string>& operations) {
		int answer = 0;
		vector<int> v;

		for (int i = 0; i < operations.size(); i++) {
			if (operations[i] == "C") {
				v.pop_back();
				continue;
			}

			if (operations[i] == "D") {
				int n = v.back();
				v.push_back(n * 2);
				continue;
			}

			if (operations[i] == "+") {
				int a = v[v.size() - 1];
				int b = v[v.size() - 2];

				v.push_back(a + b);
				continue;
			}

			int n = stoi(operations[i]);
			v.push_back(n);
		}

		for (int i = 0; i < v.size(); i++) {
			answer += v[i];
		}

		return answer;
	}
};

File: LeetCode/java/682_Baseball_Game.java
class Solution {
    public int calPoints(String[] operations) {
        int answer = 0;
        List<Integer> list = new ArrayList<>();

        for (int i = 0; i < operations.length; i++) {
            if (operations[i].equals("C")) {
                list.remove(list.size() - 1);
                continue;
            }

            if (operations[i].equals("D")) {
                int n = list.get(list.size() - 1);
                list.add(n * 2);
                continue;
            }

            if (operations[i].equals("+")) {
                int a = list.get(list.size() - 1);
                int b = list.get(list.size() - 2);

                list.add(a + b);
                continue;
            }

            int n = Integer.parseInt(operations[i]);
            list.add(n);
        }

        for (int i = 0; i < list.size(); i++) {
            answer += list.get(i);
        }

        return answer;
    }
}